scripts/generate_activity_script/src/script.ts

```diff
@@ -346,29 +325,492 @@ async function withdrawRandomNFT(
+          timeoutHeight: "0",
+          timeoutTimestamp: String(
+            (Math.floor(Date.now() / 1000) + 10 * 60) * 1e9
+          ),
+        },
+      },
+    ],
+    {
+      gas: "200000",
+      amount: [fee],
+    },
+    ""
+  );
+
+  return transactionHash;
+}
+
+/* Function dedicated to do IBC transfer from Neutron to Remote Chain
+ * Function doesn't handle any exceptions from the inner call
+ */
+async function IBCToTransfer(
+  clientSG: SigningStargateClient,
+  addressFrom: string,
+  addressTo: string,
+  channel: string,
+  port: string,
+  amount: Coin
+) {
+  const txHash = await IBCTransfer(
+    clientSG,
+    addressFrom,
+    addressTo,
+    channel,
+    port,
+    amount,
+    {
+      denom: "untrn",
+      amount: "2000",
+    }
+  );
+  return {
+    mode: NeutronAction.PROCESS_LSM_SHARES_IBC_TO,
+    txHash: txHash,
+  };
+}
+
+/* Function dedicated to choose the random amount of tokens that we want to send in processLSMShares function
+ * MAX_LSM_PROCESS from .env is the maximum amount of tokens that this function can choose to send
+ * If current balance lower than MAX_LSM_PROCESS, then MAX_LSM_PROCESS is current balance
+ */
+async function randomIBCToTransfer(
+  dropCore: DropCoreClient,
+  neutronWallet: Wallet,
+  addressFrom: string,
+  addressTo: string
+): Promise<ActionLog> {
+  const baseDenomBalance: Coin = await neutronWallet.clientCW.getBalance(
+    addressFrom,
+    BASE_DENOM
+  );
+  /* If here is nothing to send on our balance, then just throw an error
+   */
+  if (Number(baseDenomBalance.amount) === 0) {
+    throw `Nothing to transfer via IBC, ${BASE_DENOM} balance is 0`;
+  }
+
+  const exchangeRate: number = Math.floor(
+    Number(await dropCore.queryExchangeRate())
+  );
+  const minExchangeRate: number = exchangeRate + 1;
+  const config = await dropCore.queryConfig();
+
+  /* min is a minimum amount of tokens that this function can randomly choose from interval
+   * It's either a config.lsm_min_bond_amount or minExchangeRate depends on
+   * What is bigger (the biggest here is a chosen minimum)
+   */
+  const min: number =
+    Number(config.lsm_min_bond_amount) < minExchangeRate
+      ? minExchangeRate
+      : Number(config.lsm_min_bond_amount);
+  if (min > Number(baseDenomBalance.amount)) {
+    throw `Nothing to send via IBC, ${BASE_DENOM} balance is lower then min(${min}) (this value either exchange rate or config.lsm_min_bond_amount)`;
+  }
+  if (min > MAX_LSM_PROCESS) {
+    throw `MAX_LSM_PROCESS lower then min(${min}) (this value either exchange rate or config.lsm_min_bond_amount)`;
+  }
+
+  /* max is a maximum amount of tokens that this function can randomly choose from interval
+   * It's either a MAX_LSM_PROCESS or current BASE_DENOM balance depends on
+   * What is less (the smaller here is a chosen maximum)
+   */
+  const max: number =
+    Number(baseDenomBalance.amount) < MAX_LSM_PROCESS
+      ? Number(baseDenomBalance.amount)
+      : MAX_LSM_PROCESS;
```

Comment:
```suggestion
  const max = Math.max(Number(baseDenomBalance.amount), MAX_LSM_PROCESS);
```
please consider renaming this variable